polyfill/lib/ecmascript.mjs

```diff
@@ -3540,216 +3540,103 @@ export function UnbalanceDateDurationRelative(years, months, weeks, days, larges
-        }
-
-        // balance months down to days
-        while (!months.isZero()) {
-          let oneMonthDays;
-          ({ relativeTo: plainRelativeTo, days: oneMonthDays } = MoveRelativeDate(
-            calendarRec,
-            plainRelativeTo,
-            oneMonth
-          ));
-          days = days.add(oneMonthDays);
-          months = months.subtract(sign);
-        }
-      }
-      break;
-    default:
-      {
-        if (years.isZero() && months.isZero() && weeks.isZero()) break;
-        if (!calendarRec) throw new RangeError('a starting point is required for balancing calendar units');
-        // balance years down to days
-        while (!years.isZero()) {
-          let oneYearDays;
-          ({ relativeTo: plainRelativeTo, days: oneYearDays } = MoveRelativeDate(
-            calendarRec,
-            plainRelativeTo,
-            oneYear
-          ));
-          days = days.add(oneYearDays);
-          years = years.subtract(sign);
-        }
-
-        // balance months down to days
-        while (!months.isZero()) {
-          let oneMonthDays;
-          ({ relativeTo: plainRelativeTo, days: oneMonthDays } = MoveRelativeDate(
-            calendarRec,
-            plainRelativeTo,
-            oneMonth
-          ));
-          days = days.add(oneMonthDays);
-          months = months.subtract(sign);
-        }
-
-        // balance weeks down to days
-        while (!weeks.isZero()) {
-          let oneWeekDays;
-          ({ relativeTo: plainRelativeTo, days: oneWeekDays } = MoveRelativeDate(
-            calendarRec,
-            plainRelativeTo,
-            oneWeek
-          ));
-          days = days.add(oneWeekDays);
-          weeks = weeks.subtract(sign);
-        }
-      }
-      break;
+      return { years, months, weeks, days };
+    case 'month': {
+      if (years === 0) return { years: 0, months, weeks, days };
+      if (!calendarRec) throw new RangeError('a starting point is required for months balancing');
+      // balance years down to months
+      const later = CalendarDateAdd(calendarRec, plainRelativeTo, new TemporalDuration(years));
+      const untilOptions = ObjectCreate(null);
+      untilOptions.largestUnit = 'month';
+      const untilResult = CalendarDateUntil(calendarRec, plainRelativeTo, later, untilOptions);
+      const yearsInMonths = GetSlot(untilResult, MONTHS);
+      return { years: 0, months: months + yearsInMonths, weeks, days };
+    }
+    case 'week': {
+      if (years === 0 && months === 0) return { years: 0, months: 0, weeks, days };
+      if (!calendarRec) throw new RangeError('a starting point is required for weeks balancing');
+      // balance years and months down to days
+      const later = CalendarDateAdd(calendarRec, plainRelativeTo, new TemporalDuration(years, months));
+      const yearsMonthsInDays = DaysUntil(plainRelativeTo, later);
+      return { years: 0, months: 0, weeks, days: days + yearsMonthsInDays };
+    }
+    default: {
```

Comment:
```suggestion
    case 'day': {
```
(it shouldn't make a difference, but is more clear and would help expose a problem if there were one)

Reply:
It's valid for Duration.p.round() and Duration.p.total() to pass time units here, but I agree it could be more clear. I'll add a comment explaining "largestUnit is 'day', or any time unit".